### DISCUSSION TITLE
How to increase DataBufferLimit

### DISCUSSION THREAD

**Discussion by apr-amboss** [provectus/kafka-ui#2988] [Q&A] [answered]
We are running into the issue listed under '[Common problems](https://github.com/provectus/kafka-ui/wiki/Common-problems#databufferlimitexception-exceeded-limit-on-max-bytes-to-buffer)' in the wiki so I guess the solution is supposed to be trivial but I have not been able to figure out how can I increase the default buffer limit using Helm chart?
Adding the below to the `values.yaml` does not seem to be a valid option. 

```
webclient:
     max-in-memory-buffer-size: 30MB
```

Log excerpt for reference:
```
 io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: org.springframework.core.io.buffer.DataBufferLimitException: Exceeded limit on max bytes to buffer : 262144
	at org.springframework.core.io.buffer.LimitedDataBufferList.raiseLimitException(LimitedDataBufferList.java:99)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	*__checkpoint ⇢ Body from GET http://kafka-connect.kafka-production:8083/connectors/english-bigquery-sink-c-1/status [DefaultClientResponse]
Original Stack Trace:
		at org.springframework.core.io.buffer.LimitedDataBufferList.raiseLimitException(LimitedDataBufferList.java:99)
		at org.springframework.core.io.buffer.LimitedDataBufferList.updateCount(LimitedDataBufferList.java:92)
		at org.springframework.core.io.buffer.LimitedDataBufferList.add(LimitedDataBufferList.java:58)
```

Please kindly point me to the documentation that I missed or share a solution that you found yourself. Any help is much appreciated.

**Comment by Haarolean** [comment, marked as answer]
Hi, please provide your config contents (values.yaml in your case)

**Reply by apr-amboss** [reply]
Hi, please see below - this is the relevant part of the config:


> kafkaui:
  name: kafka-ui
  replicaCount:
    min: 1
    max: 3
  image:
    repository: provectuslabs/kafka-ui
    tag: latest
    pullPolicy: IfNotPresent
  env: {}
  resources:
    requests:
      cpu: 10m
      memory: 200Mi
  service:
    port: 8080
    type: ClusterIP
  webclient:
    max-in-memory-buffer-size: 30MB

**Reply by Haarolean** [reply]
Sorry, that's not how it works. That's an environment variable, in your case it's declared as a helm option or something.
Please take a look at the [helm guide](https://github.com/provectus/kafka-ui/blob/master/helm_chart.md), you have to use a configmap.

**Reply by apr-amboss** [reply]
Thanks for the guide!
So I have to use an environment variable in a ConfigMap and simply include this property? Like below?

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: kafka-ui-helm-values
data:
  KAFKA_CLUSTERS_0_NAME: "kafka-cluster-name"
  KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS: "kafka-cluster-broker-endpoints:9092"
  AUTH_TYPE: "DISABLED"
  MANAGEMENT_HEALTH_LDAP_ENABLED: "FALSE" 
  MAX-IN-MEMORY-BUFFER-SIZE: ??
```

**Reply by Haarolean** [reply]
@apr-amboss this (`webclient_max-in-memory-buffer-size`) should work :)

**Reply by apr-amboss** [reply]
Thanks, I'll give it a try!

**Reply by deiga** [reply]
@Haarolean 
Would this be the right way to do it with the yaml configmap?
```
kafka-ui:
	...
	yamlApplicationConfig:
	    webclient:
	      max-in-memory-buffer-size: 30MB
	    auth:
	      type: disabled
	    management:
	      health:
	        ldap:
	          enabled: false
  ```

**Reply by Haarolean** [reply]
@deiga sorry for the delay. Yep, it is.
